server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -49,21 +50,42 @@ protected boolean searchWithCollector(
         boolean hasFilterCollector,
         boolean hasTimeout
     ) throws IOException {
-        return searchWithCollectorManager(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
+        // create the top docs collector
+        final TopDocsCollectorContext topDocsFactory = createTopDocsCollectorContext(searchContext, hasFilterCollector);
+        return searchWithCollector(searchContext, searcher, query, collectors, topDocsFactory, hasFilterCollector, hasTimeout);
+    }
```

Comment:
It looks like this is identical as what is in the method being overridden here?

Reply:
right, before this change we always create TopDocsCollector inside the `searchWithCollectorManager`. In some scenarios with don't need that collector (we are adding custom one instead), but because it's added it consumes resources. For detailed profiling info please check main GH issue.

Reply:
I guess what I meant is why do we need this override at all then since we can use the `DefaultQueryPhaseSearcher` implementation in both cases? Can we just delete this method from `ConcurrentQueryPhaseSearcher`?

Reply:
discussed offline, I missed the point that Concurrent QPS extends the DefaultQPS, so the existing override is same between parent and child classes, thus it's not needed. Remove the override method from ConcurrentQPS